### PR TITLE
add pool for pubsub connections

### PR DESCRIPTION
**Pull request by bbaldino**
Based on discussion in https://github.com/djc/bb8/issues/70

### CODE REVIEW

File: redis/src/lib.rs [hunk 1]
```diff
@@ -39,9 +39,50 @@ pub use bb8;
...
+impl bb8::ManageConnection for RedisPubSubConnectionManager {
+    type Connection = PubSub;
+    type Error = RedisError;
+
+    async fn connect(&self) -> Result<Self::Connection, Self::Error> {
+        Ok(self.client.get_async_connection().await?.into_pubsub())
+    }
+
+    async fn is_valid(&self, _conn: &mut Self::Connection) -> Result<(), Self::Error> {
+        // TODO:
```
Comment (bbaldino) [inline redis/src/lib.rs:72]: Something needs to be done here, but not sure what would be best for a PubSub.

Comment (djc) [reply redis/src/lib.rs:72]: Maybe `punsubscribe()` with an empty pattern?

Comment (bbaldino) [reply redis/src/lib.rs:72]: Good idea.  Made that change, just trying to validate a few things.

**Comment by bbaldino** [comment]
Ok, pushed the logic for `is_valid`.  Unfortunately I wasn't able to really validate some things in my use case (due to other issues), and didn't see any unit tests in the redis that I could expand on.   If there's anything you think might be risky here (bug-wise) we can hold off.

**Comment by djc** [comment]
> Ok, pushed the logic for `is_valid`. Unfortunately I wasn't able to really validate some things in my use case (due to other issues), and didn't see any unit tests in the redis that I could expand on. If there's anything you think might be risky here (bug-wise) we can hold off.

I mean, if you can't validate this stuff, why are we considering this for inclusion?

**Comment by bbaldino** [comment]
Sorry for the lack of follow-up here...we ended up going in a different direction (no longer using redis due to other reasons), so I likely won't have time to come back to this.

**Pull request closed without merging**